jetty-ee10/jetty-ee10-websocket/jetty-ee10-websocket-jetty-server/src/main/java/org/eclipse/jetty/ee10/websocket/server/JettyWebSocketServerContainer.java

```diff
@@ -218,29 +217,24 @@ public boolean upgrade(JettyWebSocketCreator creator, HttpServletRequest request
         ServletContextResponse servletContextResponse = servletContextRequest.getServletContextResponse();
 
         WebSocketNegotiator negotiator = WebSocketNegotiator.from(coreCreator, frameHandlerFactory);
-        Handshaker handshaker = webSocketMappings.getHandshaker();
 
-        FutureCallback callback = new FutureCallback();
-        try
-        {
-            // Set the wrapped req and resp as attributes on the ServletContext Request/Response, so they
-            // are accessible when websocket-core calls back the Jetty WebSocket creator.
-            servletContextRequest.setAttribute(WebSocketConstants.WEBSOCKET_WRAPPED_REQUEST_ATTRIBUTE, request);
-            servletContextRequest.setAttribute(WebSocketConstants.WEBSOCKET_WRAPPED_RESPONSE_ATTRIBUTE, response);
+        // Set the wrapped req and resp as attributes on the ServletContext Request/Response, so they
+        // are accessible when websocket-core calls back the Jetty WebSocket creator.
+        servletContextRequest.setAttribute(WebSocketConstants.WEBSOCKET_WRAPPED_REQUEST_ATTRIBUTE, request);
+        servletContextRequest.setAttribute(WebSocketConstants.WEBSOCKET_WRAPPED_RESPONSE_ATTRIBUTE, response);
 
-            if (handshaker.upgradeRequest(negotiator, servletContextRequest, servletContextResponse, callback, components, customizer))
-            {
+        try (Blocker.Callback callback = Blocker.callback())
+        {
+            boolean upgraded = webSocketMappings.upgrade(negotiator, servletContextRequest, servletContextResponse, callback, customizer);
```

Comment:
`upgraded` is the wrong name for this, upgrade could return true but be sending and error response.